tests/acceptance/response_test.py

```diff
@@ -61,37 +66,45 @@ def get_registry(settings):
     return registry
 
 
-def get_swagger_schema(schema_dir='tests/sample_schemas/good_app/'):
-    return compile_swagger_schema(
-        schema_dir,
-        get_resource_listing(schema_dir, False)
-    )
-
-
-def _validate_against_tween(request, response=None, **overrides):
+def _validate_against_tween(request, response=None, path_pattern='/',
+                            **overrides):
     """
-    Acceptance testing helper for testing the validation tween with Swagger 1.2
+    Acceptance testing helper for testing the swagger tween with Swagger 2.0
     responses.
 
-    :param request: pytest fixture
+    :param request: pyramid request
     :param response: standard fixture by default
+    :param path_pattern: Path pattern eg. /foo/{bar}
+    :param overrides: dict of overrides for `pyramid_swagger` config
     """
     def handler(request):
         return response or Response()
 
     settings = dict({
-        'pyramid_swagger.swagger_versions': ['1.2'],
+        'pyramid_swagger.schema_directory': 'tests/sample_schemas/good_app/',
         'pyramid_swagger.enable_swagger_spec_validation': False,
-        'pyramid_swagger.schema_directory': 'tests/sample_schemas/good_app/'},
+        'pyramid_swagger.enable_response_validation': True,
+        'pyramid_swagger.swagger_versions': ['2.0']},
         **overrides
     )
-    settings['pyramid_swagger.schema12'] = get_swagger_schema()
-    settings['pyramid_swagger.schema20'] = None
+
+    spec = get_swagger_spec(settings)
+    settings['pyramid_swagger.schema12'] = None
+    settings['pyramid_swagger.schema20'] = spec
+
     registry = get_registry(settings)
 
-    # Let's make request validation a no-op so we can focus our tests.
-    with mock.patch.object(pyramid_swagger.tween, 'validate_request'):
-        validation_tween_factory(handler, registry)(request)
+    # This is a little messy because the current flow of execution doesn't
+    # set up the route_info in pyramid. Have to mock out the `route_info`
+    # so that usages in the tween meet expectations. Holler if you know a
+    # better way to do this!
+    op = spec.get_op_for_request(request.method, path_pattern)
+    mock_route_info = {'match': request.matchdict, 'route': None}
+    mock_route_mapper = Mock(spec=IRoutesMapper, return_value=mock_route_info)
+    with patch('pyramid_swagger.tween.get_op_for_request', return_value=op):
+        with patch('pyramid.registry.Registry.queryUtility',
+                   return_value=mock_route_mapper):
+            validation_tween_factory(handler, registry)(request)
 
 
 def test_response_validation_enabled_by_default():
```

Comment:
Not your problem since this code is preexisting, but just for the record, I'm not a big fan of this test existing on its own. It should be a part of a series that tests validation passes with a valid response, then fails, then test enabled vs disabled. Also there should be wrappers around the test requests.

Anyways.... _gets off soapbox_